### PR TITLE
[RFC][PoC] Core form extensions

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | yes |
| Related tickets | concerning https://github.com/Sylius/Sylius/pull/6354#pullrequestreview-3311622 |
| License | MIT |

<del>**Note**: Description to be updated, no time at hand atm.</del><br/>

Lets face it. Extending forms is messy.
Sometime all you have to do is override the service key, while on other occasions it's not that straight forward and you spend an hour searching for reasons why it doesn't work.
There's also the tagging along all previous `__construct` arguments so it won't blow up.

In most simplistic way, an ExtensionType lets us plug into the pointed form and add/remove stuff.
Same way extending does, but without all the fuss.
Plus, one form can have few extensions at once, so adding stuff later on isn't really a problem.
And it's easy as all hell.

So in short the concept is to change **all** endpoint form types into extension types and finalise the bases.
Only one's that can't be morphed are the ones that diverge on their way from the bundle, to core, User => AdminUser/ShopUser etc.

This approach would eradicate the need to override forms in almost any use case.
For those few hardcore cases that it won't work, you can still create a new form type and swap it with the default 🙈 

Cheers 🍻

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
I am 👍 , but it is too late to change the way we override forms, let's gather some feedback from the Beta and if there are problems with our current approach then we can consider this change with all its + and -. Thanks Jasiek! :)

**Comment by NeverResponse** [comment]
Closing cause of #6896 merging.
